problem-spring-web-starter/pom.xml

```diff
@@ -10,18 +10,11 @@
     </parent>
 
     <artifactId>problem-spring-web-starter</artifactId>
-    <packaging>pom</packaging>
 
     <name>Problem: Spring Web Starter</name>
     <description>Spring starter module for Problem Spring Web</description>
 
     <dependencies>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-web</artifactId>
-            <version>${spring-boot.version}</version>
-        </dependency>
```

Comment:
this works. the only problem I had with this is that the [spring doc](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-custom-starter-module-starter) seems to imply not to do this:

> Do not make assumptions about the project in which your starter is added. If the library you are auto-configuring typically requires other starters, **mention them as well**. 

In this context, it is fair to assume that if you're using spring-boot, you should be using the spring-boot-starter-web since you are using webmvc to use the Problem lib.
Apart from that, this looks good to me

Reply:
I could also dig through all the conflicting dependencies and resolve them one by one. I was just too lazy right now.

Reply:
i tried to do that this morning but i wasn't sure how to solve the conflict between `org.apache.tomcat.embed:tomcat-embed-core` and `org.glassfish:javax.servlet` since the former is one of the most important dependencies of the web-starter so excluding that one is like excluding the whole web-starter, but the second one is a dependency that comes from the zalando parent

Reply:
definitely removing the whole starter-web dependency is the easiest. I know I'm not depending on `problem-spring-web-starter` to pull `spring-boot-starter-web` transitively for me. I was just considering that spring recommended it for an unknown reason to me

Reply:
actually i ran into a case yesterday that made me think that the best way is to indeed do what you propose here: remove the starter dependency completely
The case was, i'm using the spring web starter but i use jetty instead of tomcat. for that, when i pull the spring web starter i exclude tomcat and add the jetty starter. With the Problem lib as it is in 0.25, the Problem starter was pulling again tomcat, so I had to end up excluding tomcat twice, once in an unexpected place (the Problem starter). Therefore, to not violate the principle of least surprise, the Problem starter should not pull spring-boot-starter-web

Reply:
That's a weird behavior of maven, to be honest. I'd expect that if you directly depend on the spring-boot-starter-web with an exclusion for tomcat then that should take precedence over the transitive spring-boot-starter-web coming from another dependency.